.github/workflows/_env.yml

```diff
@@ -136,21 +136,11 @@ jobs:
     - uses: actions/checkout@v4
       name: Checkout Envoy repository
       with:
-        fetch-depth: ${{ ! (inputs.check_mobile_run || inputs.trusted) && 1 || 0 }}
+        fetch-depth: ${{ ! (inputs.check_mobile_run || ! startsWith(github.event_name, 'pull_request')) && 1 || 0 }}
```

Comment:
How about the event triggered by the trusted bots?

Reply:
the condition is a bit convoluted - but it would match - ie `not startswith pull_request` == fetch_depth=0

Reply:
open to suggestions on how to make it clearer - with these pseudo ternaries your options are a bit limited

Reply:
Sorry, what I actually want to say is `how about the event triggered by the untrusted bots`. Seems that events also be treated as trusted.

Reply:
i see - at this point we havent derived the trusted state and the repo hasnt been checked out

re untrusted bots - they should never happen - but in the case they do i think this does the right thing

what we dont want is for them to be able to check out a specified commit

in this case they will get the head of the branch, in the next step they will be deemed untrusted, and the step after that they wont be able to rewind the commit

Reply:
just realizing tho - this is still not quite right - it needs to call the env action a second time i think

imagining a ci run to actually release, where the branch has been reopened by the time this gets triggered (ie not this actual workflow - but others that reuse _env.yml etc)

in this case it will trigger the run, checkout HEAD, derive env vars, but because the branch is reopened it will see the version as being `dev` - so i think we need to call the env action a second time with the correct commit

Reply:
hmm - that scheme wont quite work - but we need something like that

Reply:
k - updated with a bit of rearrangement - hopefully should work and will get/set correct commits at right times